src/components/changeUserData/ChangeEmail.tsx

```diff
@@ -0,0 +1,139 @@
+import * as Yup from 'yup';
+import axios from 'axios';
+import { Dispatch, SetStateAction, useContext } from 'react';
+import { ToastContentProps, toast } from 'react-toastify';
+import { useForm } from 'react-hook-form';
+import { useParams } from 'react-router-dom';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+import classes from '../../pages/Form.module.css';
+import { AuthContext } from '../../context/auth-context';
+
+interface UserFormValues {
+  email: string | null;
+}
+const UserFormSchema = () =>
+  Yup.object({
+    email: Yup.string()
+      .email('Invalid email.')
+      .required('Your email is required.'),
+  });
+
+const ChangeEmail = ({
+  isEditingEmail,
+  setIsEditingEmail,
+}: {
+  isEditingEmail: boolean;
+  setIsEditingEmail: Dispatch<SetStateAction<boolean>>;
+  userDataEmail: string | null;
+}) => {
+  const { token } = useContext(AuthContext);
+  const paramsUserId = useParams().userId;
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<UserFormValues>({
+    resolver: yupResolver(UserFormSchema()),
+  });
+
+  const onSubmitEmail = handleSubmit(async (data: UserFormValues) => {
+    const response = await toast.promise(
+      axios.patch(
+        `${process.env.REACT_APP_BACKEND_URL}/user/${paramsUserId}/email`,
+        data,
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        },
+      ),
+      {
+        pending: 'Please, wait.',
+        success: {
+          render() {
+            setIsEditingEmail(false);
+            // fetchUserData();
```

Comment:
After email/username changed form, UI goes back to main view will all data, but the value of changed field is still the same. I need to refresh page to see result, so refetch is needed for userData (or you can go another route to have "optimistic UI" and update immediately.
I prefer pessimistic UI, but choose whatever you want :)